### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by 4lch4**
- Removed trailing whitespaces.
- Fixed a typo where `components` was misspelled.
- Moved all links to references at the bottom of the file.
- On line 162, I added bash for the file tree example as it was used for the previous example on line 29.
- Added a new line between examples that had a DO and DO NOT.
  - The code snippet from line 85 is a good example. Having that li'l space made it much nicer to read.

### CODE REVIEW

File: docs/src/pages/guides/publish-to-npm.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ---
 layout: ~/layouts/MainLayout.astro
 title: Publish to NPM
+modifiedDate: '2021-09-06 @ 11:28:21'
```
Comment (FredKSchott) [inline docs/src/pages/guides/publish-to-npm.md:4]: curious, what's the reason for adding this?

File: docs/src/pages/guides/publish-to-npm.md [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 ---
 
-Building a new Astro component? **Publish it to [npm!](https://npmjs.com/)**
+Building a new Astro component? **Publish it to [npm!][NPM]**
```
Comment (FredKSchott) [inline docs/src/pages/guides/publish-to-npm.md:7]: I actually personally prefer inline URLs, but I have started to see how this makes future refactoring much better, which is a pretty nice reason to use them. I'll swallow my personal pref and go +1 for doing this here and across the site moving forward

**Comment by FredKSchott** [comment]
Awesome! cherry-picked your commit and merged into #1323 with some minor changes. Thanks for help!

File: docs/src/pages/guides/publish-to-npm.md (hunk 1 above)
Comment (4lch4) [reply docs/src/pages/guides/publish-to-npm.md:4]: Whoops.. So sorry for the delay, this was auto-added by an extension of mine and I missed it 🤐

File: docs/src/pages/guides/publish-to-npm.md (hunk 2 above)
Comment (4lch4) [reply docs/src/pages/guides/publish-to-npm.md:7]: I wasn't sure where you leaned so I'm glad it worked out for the best ^_^

**Pull request closed without merging**